pkg/eventbus/listener/normalize/listener.go

```diff
@@ -0,0 +1,193 @@
+/*
+Copyright 2021 Loggie Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package normalize
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/loggie-io/loggie/pkg/eventbus/export/logger"
+	promeExporter "github.com/loggie-io/loggie/pkg/eventbus/export/prometheus"
+	"github.com/prometheus/client_golang/prometheus"
+	"time"
+
+	"github.com/loggie-io/loggie/pkg/core/api"
+	"github.com/loggie-io/loggie/pkg/core/log"
+	"github.com/loggie-io/loggie/pkg/eventbus"
+)
+
+const name = "normalize"
+
+func init() {
+	eventbus.Registry(name, makeListener, eventbus.WithTopics([]string{eventbus.NormalizeTopic, eventbus.ComponentBaseTopic}))
+}
+
+func makeListener() eventbus.Listener {
+	l := &Listener{
+		done:                 make(chan struct{}),
+		config:               &Config{},
+		eventChan:            make(chan eventbus.Event),
+		normalizeMetric:      make(map[string]map[string]*eventbus.NormalizeMetricData),
+		nameComponentMetrics: make(map[string][]eventbus.ComponentBaseMetricData),
+	}
+	return l
+}
+
+type Config struct {
+	Period           time.Duration `yaml:"period" default:"30s"`
```

Comment:
it's better to set default period to 10s?

Reply:
ok